src/loader/bgw_launcher.c

```diff
@@ -416,12 +441,16 @@ message_start_action(HTAB *db_htab, BgwMessage *message, VirtualTransactionId vx
 	 */
 	if (!found)
 	{
+		db_he->db_scheduler_handle = NULL;
+		db_he->incremented = false;
+
```

Comment:
So, this seems to be the fundamental issue here.

What should happen is that the number of hash table entries SHOULD reflect how many times this launcher has incremented the counter. But the issue seems to have been that we sometimes created a new slot and then failed to increment, thus ending up with a garbage slot that was never accounted for but later caused a decrement in the garbage collector.

What we should do is either delete the slot (`HASH_REMOVE`) on failure to increment, or (preferably) always try to increment first and then do `HASH_ENTER`. If we find an existing slot (`found == true`), then decrement since that slot was already accounted for.

Thus, the simple rule should be that the number of hash table entries should always reflect the number of increments the launcher has done to the counter.

This stuff with the `incremented` bool seems unnecessary to me.